File: it-code/nodejs-code-step/util/Array.js
function log(e){console.log(e)}
function print(e){console.log(e)}
arr = [1,2,3]
for(key in arr) print(key);
for(value of arr) print(value);
print({
  "init arr ": [1,2,3,"a","b","c"],
  "reverse": Array(3).fill(1).map((e,i)=>i).reverse(),
  "slice": [1,2,3].slice(1) ,
  "splice": [1,2,3].splice(1,1,2) ,
  "concat":[].concat([1,2,3],[4,5,6]),
  "delete": arr.splice(0,1),
})
print(arr)

File: it-code/python-code-step/util/Array.py
# -*- coding: utf-8 -*-
import numpy as np
print(''' 
test array
''')
print({
  "init arr ": [1,2,3,"a","b","c"],
  "reverse": list(range(3))[::-1],
  "slice": list(range(3))[0:1],
  "concat":np.append([1,2,3],[4,5,6]),
})      

a = np.array([1,2,3,4,5])
slice = a[:3]
slice[0] = 100
print(a)